pkg/collector/corechecks/ebpf/probe/oom_kill.go

```diff
@@ -18,46 +25,76 @@ import (
 */
 import "C"
 
+const oomMapName = "oom_stats"
+
 type OOMKillProbe struct {
-	m      *bpflib.Module
-	oomMap *bpflib.Table
+	m      *manager.Manager
+	oomMap *bpflib.Map
 }
 
 func NewOOMKillProbe(cfg *ebpf.Config) (*OOMKillProbe, error) {
-	source, err := ebpf.PreprocessFile(cfg.BPFDir, "oom-kill-kern.c")
+	compiledOutput, err := runtime.OomKill.Compile(cfg, nil)
 	if err != nil {
-		return nil, fmt.Errorf("Couldn’t process headers for asset “oom-kill-kern.c”: %v", err)
+		return nil, err
 	}
+	defer compiledOutput.Close()
 
-	m := bpflib.NewModule(source.String(), []string{})
-	if m == nil {
-		return nil, fmt.Errorf("failed to compile “oom-kill-kern.c”")
+	probes := []*manager.Probe{
+		&manager.Probe{
+			Section: "kprobe/oom_kill_process",
+		},
 	}
 
-	kprobe, err := m.LoadKprobe("kprobe__oom_kill_process")
-	if err != nil {
-		return nil, fmt.Errorf("failed to load kprobe__oom_kill_process: %s\n", err)
+	maps := []*manager.Map{
+		{Name: "oom_stats"},
+	}
+
+	m := &manager.Manager{
+		Probes: probes,
+		Maps:   maps,
+	}
+
+	managerOptions := manager.Options{
+		RLimit: &unix.Rlimit{
+			Cur: math.MaxUint64,
+			Max: math.MaxUint64,
+		},
 	}
 
-	if err := m.AttachKprobe("oom_kill_process", kprobe, -1); err != nil {
-		return nil, fmt.Errorf("failed to attach oom_kill_process: %s\n", err)
+	if err := m.InitWithOptions(compiledOutput, managerOptions); err != nil {
+		return nil, errors.Wrap(err, "failed to init manager")
 	}
 
-	table := bpflib.NewTable(m.TableId("oomStats"), m)
+	if err := m.Start(); err != nil {
+		return nil, errors.Wrap(err, "failed to start manager")
+	}
+
+	oomMap, ok, err := m.GetMap(oomMapName)
+	if err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, fmt.Errorf("failed to get map '%s'", statsMapName)
+	}
 
 	return &OOMKillProbe{
 		m:      m,
-		oomMap: table,
+		oomMap: oomMap,
 	}, nil
 }
 
 func (k *OOMKillProbe) Close() {
-	k.m.Close()
+	k.m.Stop(manager.CleanAll)
 }
 
 func (k *OOMKillProbe) GetAndFlush() []OOMKillStats {
 	results := k.Get()
-	k.oomMap.DeleteAll()
+	var pid uint32
+	it := k.oomMap.Iterate()
+	for it.Next(&pid, nil) {
```

Comment:
I think you also need to `it.Err()` after the loop.

Reply:
I believe the `.Next` and `.Delete` calls need to wrap `&pid` in `unsafe.Pointer`.

Reply:
uint32 is a known type to the library

Reply:
It will marshal the value in/out though, which is less performant than just passing the pointer directly.